In [172]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
#import seaborn as sns

In [173]:

df=pd.read_csv(r'C:\AMAZONPROJECT\data\amazon.csv')
df

product_id                                       product_name  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category discounted_price  \
0     Computers&Accessories|Accessories&Peripherals|...             ₹399   
1     Computers&Accessories|Accessories&Peripherals|...             ₹199   
2     Computers&Accessories|Accessories&Peripherals|...             ₹199   
3     Computers&Accessories|Accessories&Peripherals|...             ₹329   
4     Computers&Accessories|Accessories&Peripherals|...             ₹154   
...                                                 ...              ...   
1460  Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...             ₹379   
1461  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...           ₹2,280   
1462  Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...           ₹2,219   
1463  Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...           ₹1,399   
1464  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...           ₹2,863   

     actual_price discount_percentage rating rating_count  \
0          ₹1,099                 64%    4.2       24,269   
1            ₹349                 43%      4       43,994   
2          ₹1,899                 90%    3.9        7,928   
3            ₹699                 53%    4.2       94,363   
4            ₹399                 61%    4.2       16,905   
...           ...                 ...    ...          ...   
1460         ₹919                 59%      4        1,090   
1461       ₹3,045                 25%    4.1        4,118   
1462       ₹3,080                 28%    3.6          468   
1463       ₹1,890                 26%      4        8,031   
1464       ₹3,690                 22%    4.3        6,987   

                                          about_product  \
0     High Compatibility : Compatible With iPhone 12...   
1     Compatible with all Type C enabled devices, be...   
2     【 Fast Charger& Data Sync】-With built-in safet...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
4     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
...                                                 ...   
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                              user_name  \
0     Manav,Adarsh gupta,Sunde

## features


In [174]:
df.shape

(1465, 16)

In [175]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype
---  ------               --------------  -----
 0   product_id           1465 non-null   str  
 1   product_name         1465 non-null   str  
 2   category             1465 non-null   str  
 3   discounted_price     1465 non-null   str  
 4   actual_price         1465 non-null   str  
 5   discount_percentage  1465 non-null   str  
 6   rating               1465 non-null   str  
 7   rating_count         1463 non-null   str  
 8   about_product        1465 non-null   str  
 9   user_id              1465 non-null   str  
 10  user_name            1465 non-null   str  
 11  review_id            1465 non-null   str  
 12  review_title         1465 non-null   str  
 13  review_content       1465 non-null   str  
 14  img_link             1465 non-null   str  
 15  product_link         1465 non-null   str  
dtypes: str(16)
memory usage: 183.3 KB


In [203]:
df['discounted_price']=df['discounted_price'].astype(str)

In [204]:
df['discounted_price']=df['discounted_price'].str.replace('₹','').str.replace(',','')
df['discounted_price']=df['discounted_price'].astype(float)

In [216]:
df['actual_price']=df['actual_price'].astype(str)

In [217]:
df['actual_price']=df['actual_price'].str.replace('₹','').str.replace(',','')
df['actual_price']=df['actual_price'].astype(float)

In [218]:
count=df['actual_price'].str.contains('-').sum()
print(count)

AttributeError: Can only use .str accessor with string values, not floating

In [219]:
df=df[~df['actual_price'].str.contains('-')]

AttributeError: Can only use .str accessor with string values, not floating

In [220]:
df['discount_percentage']=df['discount_percentage'].str.replace('%','')
df['discount_percentage']=df['discount_percentage'].astype(float)/100


AttributeError: Can only use .str accessor with string values, not floating

In [221]:
df['rating_count']=df['rating_count'].str.replace(',','')
df['rating_count']=df['rating_count'].astype(float)/100


AttributeError: Can only use .str accessor with string values, not floating

In [222]:
df['rating']= pd.to_numeric(df['rating'],errors='coerce')

In [201]:
df['rating'].isna().sum()

np.int64(0)

In [202]:
df=df.dropna(subset=['rating'])

In [223]:
df.dtypes

product_id                 str
product_name               str
category                   str
discounted_price       float64
actual_price           float64
discount_percentage    float64
rating                 float64
rating_count           float64
about_product              str
user_id                    str
user_name                  str
review_id                  str
review_title               str
review_content             str
img_link                   str
product_link               str
dtype: object

In [142]:
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category  discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             399.0   
1  Computers&Accessories|Accessories&Peripherals|...             199.0   
2  Computers&Accessories|Accessories&Peripherals|...             199.0   
3  Computers&Accessories|Accessories&Peripherals|...             329.0   
4  Computers&Accessories|Accessories&Peripherals|...             154.0   

   actual_price  discount_percentage  rating  rating_count  \
0        1099.0                 0.64     4.2        242.69   
1         349.0                 0.43     4.0        439.94   
2        1899.0                 0.90     3.9         79.28   
3         699.0                 0.53     4.2        943.63   
4         399.0                 0.61     4.2        169.05   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

In [143]:
df.describe()

discounted_price   actual_price  discount_percentage       rating  \
count       1459.000000    1459.000000          1459.000000  1459.000000   
mean        3135.912563    5463.896696             0.477197     4.101645   
std         6956.458900   10893.054833             0.216300     0.331638   
min           39.000000      39.000000             0.000000     2.000000   
25%          325.000000     825.000000             0.320000     4.000000   
50%          799.000000    1690.000000             0.500000     4.100000   
75%         1999.000000    4331.480000             0.630000     4.300000   
max        77990.000000  139900.000000             0.940000     9.600000   

       rating_count  
count   1457.000000  
mean     183.611469  
std      428.292786  
min        0.020000  
25%       11.930000  
50%       51.950000  
75%      173.940000  
max     4269.730000

In [144]:

string_columns = df.select_dtypes(include=['object']).columns
for col in string_columns:
    df[col] = df[col].str.strip()

C:\Users\Sameh\AppData\Local\Temp\ipykernel_9364\2644238989.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  string_columns = df.select_dtypes(include=['object']).columns


In [145]:
df.duplicated().sum()

np.int64(0)

In [146]:
df.duplicated(subset=['product_id']).sum()

np.int64(114)

In [147]:
df.duplicated(subset=['product_name']).sum()

np.int64(128)

In [148]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [149]:
df=df.dropna(subset=['rating','rating_count'])

In [150]:

invalid_prices = df[df['discounted_price'] > df['actual_price']]
print(f"number of invalid prices {len(invalid_prices)}")


#df = df[df['discounted_price'] <= df['actual_price']]

number of invalid prices 0


### Descriptive Statistics

1) Summary Statistics

In [151]:
df.columns

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link'],
      dtype='str')

In [152]:
df.describe()

discounted_price   actual_price  discount_percentage       rating  \
count       1457.000000    1457.000000          1457.000000  1457.000000   
mean        3139.909698    5470.025587             0.476788     4.101784   
std         6960.397205   10899.275893             0.216165     0.329768   
min           39.000000      39.000000             0.000000     2.000000   
25%          326.000000     825.000000             0.320000     4.000000   
50%          799.000000    1695.000000             0.500000     4.100000   
75%         1999.000000    4332.960000             0.630000     4.300000   
max        77990.000000  139900.000000             0.940000     9.600000   

       rating_count  
count   1457.000000  
mean     183.611469  
std      428.292786  
min        0.020000  
25%       11.930000  
50%       51.950000  
75%      173.940000  
max     4269.730000

In [153]:
df.isna().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           0
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

### Insights about pricing :
prices mean = 5463  
prices std  = 10893  
* actual_price has right skew  
* prices are very spread out
* Some items are extremely expensive

### Insights about discount_percentage :

* discount_percentage is balanced  
* no items have a huge discount than other

### Insights about rating :



In [99]:
df['rating'].median()

np.float64(4.1)

In [ ]:
df['rating'].skew()

np.float64(2.542604677475101)

rating mean = 4.1    
median      = 4.1  
std         = 0.3    
* most ratings are close to 4 , means most peole give simmilar rating
min = 2  
max = 9  
* most ratings are around 4 , with some low and high extremes

### Insights about rating count :



In [ ]:
#number of products
df['product_name'].count()

np.int64(1459)

In [ ]:
# number of ratings
df['rating_count'].count()

np.int64(1457)

In [ ]:
df['rating_count'].median()

np.float64(51.95)

1) all the products have been rated from coustmers , excecpt 2 products  

2) mean = 183  
  median = 51  
  std = 428  
   means that ther's some products have very high number of ratings , may reflect:  
    - high popularity 
    - high number of selling of this product  
    - sign to best sellers products

# Products outliers

In [ ]:

Q1 = df['rating_count'].quantile(0.25)
Q3= df['rating_count'].quantile(0.75)
IQR = Q3-Q1
extreme_products = df[df['rating_count'] > Q3 + 1.5*IQR]
extreme_products[['product_name' , 'rating_count']].sort_values('rating_count' , ascending = False)
extreme_products

product_id                                       product_name  \
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
8     B008IFXQFU  TP-Link USB WiFi Adapter for PC(TL-WN725N), N1...   
9     B082LZGK39  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
11    B0789LZTCJ  boAt Rugged v3 Extra Tough Unbreakable Braided...   
...          ...                                                ...   
1032  B01GFTEV5Y  Pigeon by Stovekraft Cruise 1800 watt Inductio...   
1045  B09X5C9VLK  Lifelong LLMG23 Power Pro 500-Watt Mixer Grind...   
1077  B008YW8M0G  Bajaj DX-7 1000W Dry Iron with Advance Solepla...   
1130  B0756K5DYZ  Prestige Iris 750 Watt Mixer Grinder with 3 St...   
1145  B0BR4F878Q  Swiffer Instant Electric Water Heater Faucet T...   

                                               category  discounted_price  \
1     Computers&Accessories|Accessories&Peripherals|...             199.0   
3     Computers&Accessories|Accessories&Peripherals|...             329.0   
8     Computers&Accessories|NetworkingDevices|Networ...             499.0   
9     Computers&Accessories|Accessories&Peripherals|...             199.0   
11    Computers&Accessories|Accessories&Peripherals|...             299.0   
...                                                 ...               ...   
1032  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            1699.0   
1045  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            1299.0   
1077  Home&Kitchen|Kitchen&HomeAppliances|Vacuum,Cle...             775.0   
1130  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            3249.0   
1145  Home&Kitchen|Heating,Cooling&AirQuality|WaterH...            1439.0   

      actual_price  discount_percentage  rating  rating_count  \
1            349.0                 0.43     4.0        439.94   
3            699.0                 0.53     4.2        943.63   
8            999.0                 0.50     4.2       1796.91   
9            299.0                 0.33     4.0        439.94   
11           799.0                 0.63     4.2        943.63   
...            ...                  ...     ...           ...   
1032        3193.0                 0.47     3.8        540.32   
1045        3500.0                 0.63     3.8        440.50   
1077         875.0                 0.11     4.2        466.47   
1130        6295.0                 0.48     3.9        430.70   
1145        1999.0                 0.28     4.8        538.03   

                                          about_product  \
1     Compatible with all Type C enabled devices, be...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
8     USB WiFi Adapter —— Speedy wireless transmissi...   
9     Universal Compatibility – It is compatible wit...   
11    The boAt rugged cable features our special tou...   
...                                                 ...   
1032  Pigeon Cruise Induction Cooktop Comes with 7 s...   
1045  3 jars liquidizing jar (1.5 liters), dry or we...   
1077  Non-stick coated golden color sole plate|Super...   
1130  Motor Warranty: 2; Product Warranty: 2;|Operat...   
1145  ✔Quick Electric Hot Water Tap Heating tube: hi...   

                                                user_id  \
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
8     AGV3IEFANZCKECFGUM42MRH5FNOA,AEBO7NWCNXKT4AESA...   
9     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
11    AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
...                                                 ...   
1032  AFVYGOA4AWO77UIPMUNH6YSKSB5A,AEJMVHQ4LOSVJQ5BO...   
1045  AFHU7KCA3ZL6XOL3PYSGYJM4LAZA,AESFKEVRHPM6BNLUQ...   
1077  AET6ITYPXTZDZO5QV36VRCTRCTVQ,AHIQYP5QKXYWXGJC5...   
1130  AENY7MQ3WUVPIJ5I5GPDPMC3NKPA,AGUCURUVW2ZY4B5WL...   
1145  AEU7DVFEL43XZ6T4D572W2ZLBRKQ,AGWOH7CFDMUVW52NM...   

                             

# Catogery outliers

In [ ]:
# highest catogries in ratingcount
extreme_categories = extreme_products['category'].value_counts()
extreme_categories

In [ ]:
# display the highest products in highst category 
extreme_products [ extreme_products['category'] == 'Electronics|Headphones,Earbuds&Accessories|Headphones|In-Ear' ]
extreme_products [ ['product_id' ,'product_name' , 'discount_percentage', 'discounted_price' ] ]

product_id                                       product_name  \
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
8     B008IFXQFU  TP-Link USB WiFi Adapter for PC(TL-WN725N), N1...   
9     B082LZGK39  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
11    B0789LZTCJ  boAt Rugged v3 Extra Tough Unbreakable Braided...   
...          ...                                                ...   
1032  B01GFTEV5Y  Pigeon by Stovekraft Cruise 1800 watt Inductio...   
1045  B09X5C9VLK  Lifelong LLMG23 Power Pro 500-Watt Mixer Grind...   
1077  B008YW8M0G  Bajaj DX-7 1000W Dry Iron with Advance Solepla...   
1130  B0756K5DYZ  Prestige Iris 750 Watt Mixer Grinder with 3 St...   
1145  B0BR4F878Q  Swiffer Instant Electric Water Heater Faucet T...   

      discount_percentage  discounted_price  
1                    0.43             199.0  
3                    0.53             329.0  
8                    0.50             499.0  
9                    0.33             199.0  
11                   0.63             299.0  
...                   ...               ...  
1032                 0.47            1699.0  
1045                 0.63            1299.0  
1077                 0.11             775.0  
1130                 0.48            3249.0  
1145                 0.28            1439.0  

[141 rows x 4 columns]

In [ ]:
# display the highest products in highest category sorted by discount_percentage
filtered = extreme_products [ extreme_products['category'] == 'Electronics|Headphones,Earbuds&Accessories|Headphones|In-Ear' ].sort_values('discount_percentage' , ascending= False)
filtered[['product_id' ,'product_name' , 'discount_percentage' , 'rating_count']]


product_id                                       product_name  \
720  B086WMSCN3  boAt Airdopes 171 in Ear Bluetooth True Wirele...   
656  B07LG59NPV  Boult Audio Probass Curve Bluetooth Wireless i...   
349  B08D77XZX5  PTron Tangentbeat in-Ear Bluetooth 5.0 Wireles...   
912  B08MTLLSL8  boAt Bassheads 102 Wired in Ear Earphones with...   
639  B08H9Z3XQW  boAt Bassheads 242 in Ear Wired Earphones with...   
609  B07TCN5VR9  Boult Audio BassBuds X1 in-Ear Wired Earphones...   
415  B085W8CFLH  PTron Tangent Lite Bluetooth 5.0 Earphones wit...   
585  B09N3ZNHTY  boAt Airdopes 141 Bluetooth Truly Wireless in ...   
618  B07KY3FNQP  boAt Bassheads 152 in Ear Wired Earphones with...   
595  B08TV2P1N8  boAt Rockerz 255 Pro+ in-Ear Bluetooth Neckban...   
352  B07GPXXNNG  boAt Bassheads 100 in Ear Wired Earphones with...   
584  B071Z8M4KX  boAt BassHeads 100 in-Ear Wired Headphones wit...   
636  B092X94QNQ  boAt Rockerz 330 in-Ear Bluetooth Neckband wit...   
400  B07GQD4K6L  boAt Bassheads 100 in Ear Wired Earphones with...   
437  B07S9S86BF  boAt Bassheads 242 in Ear Wired Earphones with...   
591  B08JQN8DGZ  boAt Airdopes 121v2 in-Ear True Wireless Earbu...   
479  B01DF26V7A  JBL C100SI Wired In Ear Headphones with Mic, J...   
716  B08FB2LNSZ  JBL Tune 215BT, 16 Hrs Playtime with Quick Cha...   
601  B07JQKQ91F  JBL C50HI, Wired in Ear Headphones with Mic, O...   
708  B07DFYJRQV  JBL C200SI, Premium in Ear Wired Earphones wit...   
347  B01DEWVZ2C  JBL C100SI Wired In Ear Headphones with Mic, J...   
598  B01DEWVZ2C  JBL C100SI Wired In Ear Headphones with Mic, J...   
760  B01MF8MB65  boAt Bassheads 225 in Ear Wired Earphones with...   
665  B07CD2BN46  Xiaomi Mi Wired in Ear Earphones with Mic Basi...   
977  B083T5G5PM  Sennheiser CX 80S in-Ear Wired Headphones with...   
930  B07XJWTYM2  realme Buds Wireless in Ear Bluetooth Earphone...   

     discount_percentage  rating_count  
720                 0.80        475.21  
656                 0.80       1030.52  
349                 0.76        581.62  
912                 0.69        760.42  
639                 0.69       1616.77  
609                 0.67        770.27  
415                 0.67        839.96  
585                 0.67       1369.54  
618                 0.65        917.70  
595                 0.65       1418.41  
352                 0.65       3637.13  
584                 0.63       3637.11  
636                 0.62       1098.64  
400                 0.62       3637.13  
437                 0.60       1616.79  
591                 0.57       1809.98  
479                 0.54       1925.89  
716                 0.50        877.98  
601                 0.50        929.95  
708                 0.47        536.48  
347                 0.40       1925.90  
598                 0.40       1925.87  
760                 0.30       2731.89  
665                 0.28       1194.66  
977                 0.25        982.50  
930                 0.16        725.63

# Relation between discount_percentage and rating_count

In [ ]:
import matplotlib.pyplot as plt
plt.xlabel = plt.matplotlib.pyplot.xlabel
plt.ylabel = plt.matplotlib.pyplot.ylabel
plt.title = plt.matplotlib.pyplot.title

In [ ]:
plt.plot(filtered['discount_percentage'], filtered['rating_count'], marker='o')
plt.xlabel('Discount percentage')
plt.ylabel('Rating count')
plt.title('Discount vs Rating Count')
plt.show()

# Top 10 products

In [249]:
df['popularity_score'] = (df['rating']*df['rating_count']/100).round(2)
df.drop('popularity',axis=1 )

product_id                                       product_name  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category  discounted_price  \
0     Computers&Accessories|Accessories&Peripherals|...             399.0   
1     Computers&Accessories|Accessories&Peripherals|...             199.0   
2     Computers&Accessories|Accessories&Peripherals|...             199.0   
3     Computers&Accessories|Accessories&Peripherals|...             329.0   
4     Computers&Accessories|Accessories&Peripherals|...             154.0   
...                                                 ...               ...   
1460  Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...             379.0   
1461  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            2280.0   
1462  Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...            2219.0   
1463  Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...            1399.0   
1464  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            2863.0   

      actual_price  discount_percentage  rating  rating_count  \
0           1099.0                 0.64     4.2        242.69   
1            349.0                 0.43     4.0        439.94   
2           1899.0                 0.90     3.9         79.28   
3            699.0                 0.53     4.2        943.63   
4            399.0                 0.61     4.2        169.05   
...            ...                  ...     ...           ...   
1460         919.0                 0.59     4.0         10.90   
1461        3045.0                 0.25     4.1         41.18   
1462        3080.0                 0.28     3.6          4.68   
1463        1890.0                 0.26     4.0         80.31   
1464        3690.0                 0.22     4.3         69.87   

                                          about_product  \
0     High Compatibility : Compatible With iPhone 12...   
1     Compatible with all Type C enabled devices, be...   
2     【 Fast Charger& Data Sync】-With built-in safet...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
4     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
...                                                 ...   
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                             

In [251]:
top_10_products = df.sort_values(['popularity_score'], ascending = False).head(10)
top_10_products[['product_id' ,'product_name' ,'category','popularity_score']]

product_id                                       product_name  \
12   B07KSMBL2H  AmazonBasics Flexible Premium HDMI Cable (Blac...   
47   B014I8SSD0  Amazon Basics High-Speed HDMI Cable, 6 Feet - ...   
684  B07KSMBL2H  AmazonBasics Flexible Premium HDMI Cable (Blac...   
65   B014I8SX4Y  Amazon Basics High-Speed HDMI Cable, 6 Feet (2...   
400  B07GQD4K6L  boAt Bassheads 100 in Ear Wired Earphones with...   
352  B07GPXXNNG  boAt Bassheads 100 in Ear Wired Earphones with...   
584  B071Z8M4KX  boAt BassHeads 100 in-Ear Wired Headphones wit...   
371  B09GFLXVH9  Redmi 9A Sport (Coral Green, 2GB RAM, 32GB Sto...   
566  B09GFPN6TP  Redmi 9A Sport (Coral Green, 3GB RAM, 32GB Sto...   
473  B09GFM8CGS  Redmi 9A Sport (Carbon Black, 2GB RAM, 32GB St...   

                                              category  popularity_score  
12   Electronics|HomeTheater,TV&Video|Accessories|C...            187.87  
47   Electronics|HomeTheater,TV&Video|Accessories|C...            187.87  
684  Electronics|HomeTheater,TV&Video|Accessories|C...            187.87  
65   Electronics|HomeTheater,TV&Video|Accessories|C...            187.87  
400  Electronics|Headphones,Earbuds&Accessories|Hea...            149.12  
352  Electronics|Headphones,Earbuds&Accessories|Hea...            149.12  
584  Electronics|Headphones,Earbuds&Accessories|Hea...            149.12  
371  Electronics|Mobiles&Accessories|Smartphones&Ba...            128.67  
566  Electronics|Mobiles&Accessories|Smartphones&Ba...            128.67  
473  Electronics|Mobiles&Accessories|Smartphones&Ba...            128.67